plugin.php

```diff
@@ -35,39 +37,91 @@ function queue_export( $config = null ) {
  * @param  string $config
  */
 function static_page_save( $config = null ) {
-
 	// Track the background task in an option for reporting / status checking.
 	$update_progress = [
 		'date'      => time(),
 		'urls'      => [],
 		'done_urls' => [],
 	];
+
 	$option_name = 'static_page_save_running';
 	update_option( $option_name, $update_progress );
 
-	$urls = get_site_urls();
+	$posts_per_page = 50;
 
-	$update_progress['urls'] = $urls;
-	update_option( $option_name, $update_progress );
+	$query_args = [
+		'post_type'              => 'any',
+		'posts_per_page'         => $posts_per_page,
+		'paged'                  => 1,
+		'fields'                 => 'ids',
+		'no_found_rows'          => true,
+		'update_post_meta_cache' => false,
+		'update_post_term_cache' => false,
+	];
+
+	$query = new WP_Query( $query_args );
+	if ( ! $query->have_posts() ) {
+		return;
+	}
+
+	$total_pages = $query->max_num_pages;
+	do {
+		$urls = get_site_urls( $config, $query_args['paged'], $posts_per_page );
+
+		if ( empty( $urls ) ) {
+			continue;
+		}
+
+		// Update progress list.
+		$update_progress         = get_option( $option_name, $update_progress );
+		$update_progress['urls'] = array_merge( $update_progress['urls'], $urls );
+		update_option( $option_name, $update_progress );
+
+		if ( ! wp_next_scheduled( 'process_static_pages', [ $config, $urls, $query_args['paged'], $total_pages ] ) ) {
+			wp_schedule_single_event( time() + 300, 'process_static_pages', [ $config, $urls, $query_args['paged'], $total_pages ] );
+		}
+
+		$query_args['paged'] ++;
+		$query = new WP_Query( $query_args );
```

Comment:
Readability: Minor nit pick - I think it's a little easier to read the `while` section with spacing.